x/ecocredit/core/msg_seal_batch.go

```diff
@@ -0,0 +1,37 @@
+package core
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+)
+
+var _ legacytx.LegacyMsg = &MsgSealBatch{}
+
+// Route implements the LegacyMsg interface.
+func (m MsgSealBatch) Route() string { return sdk.MsgTypeURL(&m) }
+
+// Type implements the LegacyMsg interface. x
+func (m MsgSealBatch) Type() string { return sdk.MsgTypeURL(&m) }
+
+// GetSignBytes implements the LegacyMsg interface.
+func (m MsgSealBatch) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ecocredit.ModuleCdc.MustMarshalJSON(&m))
+}
+
+// ValidateBasic does a sanity check on the provided data.
+func (m *MsgSealBatch) ValidateBasic() error {
```

Comment:
this message is missing a field, either the uint64 batch id or the batch denom string

Reply:
@robert-zaremba are we addressing this issue here or in follow up PR? ^

Reply:
Nice catch. This will be needed when implementing the message server methods. I think it would be ok to address as part of #973.

Reply:
Yes, adding it now.

Reply:
I've added `batch_denom`